handler.go

```diff
@@ -0,0 +1,434 @@
+	}
+}
+
+// WithCallerFlags can be used to overwrite the default caller flag option.
+func WithCallerFlags(flags uint32) HandlerOption {
+	return func(opts *handlerOpts) {
+		opts.flag = flags
+	}
+}
+
+// WithTimeSource can be used to overwrite the time sourced from the slog
+// Record.
+func WithTimeSource(fn func() time.Time) HandlerOption {
+	return func(opts *handlerOpts) {
+		opts.timeSource = fn
+	}
+}
+
+// WithNoTimestamp is an option that can be used to omit timestamps from the log
+// lines.
+func WithNoTimestamp() HandlerOption {
+	return func(opts *handlerOpts) {
+		opts.withTimestamp = false
+	}
+}
+
+// DefaultHandler is a Handler that can be used along with NewSLogger to
+// instantiate a structured logger.
+type DefaultHandler struct {
+	opts *handlerOpts
+
+	level           int64
+	tag             string
+	fields          []slog.Attr
+	callstackOffset bool
+
+	flag uint32
+	buf  *buffer
+	mu   *sync.Mutex
+	w    io.Writer
+}
+
+// A compile-time check to ensure that DefaultHandler implements Handler.
+var _ Handler = (*DefaultHandler)(nil)
+
+// Level returns the current logging level of the Handler.
+//
+// NOTE: This is part of the Handler interface.
+func (d *DefaultHandler) Level() Level {
+	return Level(atomic.LoadInt64(&d.level))
+}
+
+// SetLevel changes the logging level of the Handler to the passed
+// level.
+//
+// NOTE: This is part of the Handler interface.
+func (d *DefaultHandler) SetLevel(level Level) {
+	atomic.StoreInt64(&d.level, int64(level))
+}
+
+// NewDefaultHandler creates a new Handler that can be used along with
+// NewSLogger to instantiate a structured logger.
+func NewDefaultHandler(w io.Writer, options ...HandlerOption) *DefaultHandler {
+	opts := defaultHandlerOpts()
+	for _, o := range options {
+		o(opts)
+	}
+
+	return &DefaultHandler{
+		w:     w,
+		level: int64(LevelInfo),
+		opts:  opts,
+		buf:   newBuffer(),
+		mu:    &sync.Mutex{},
+	}
+}
+
+// Enabled reports whether the handler handles records at the given level.
+//
+// NOTE: this is part of the slog.Handler interface.
+func (d *DefaultHandler) Enabled(_ context.Context, level slog.Level) bool {
+	return atomic.LoadInt64(&d.level) <= int64(level)
+}
+
+// Handle handles the Record. It will only be called if Enabled returns true.
+//
+// NOTE: this is part of the slog.Handler interface.
+func (d *DefaultHandler) Handle(_ context.Context, r slog.Record) error {
+	buf := newBuffer()
+	defer buf.free()
+
+	// Timestamp.
+	if d.opts.withTimestamp {
+		// First check if the options provided specified a different
+		// time source to use. Otherwise, use the provided record time.
+		if d.opts.timeSource != nil {
+			writeTimestamp(buf, d.opts.timeSource())
+		} else if !r.Time.IsZero() {
+			writeTimestamp(buf, r.Time)
+		}
+	}
+
+	// Level.
+	d.writeLevel(buf, Level(r.Level))
+
+	// Sub-system tag.
+	if d.tag != "" {
+		buf.writeString(" " + d.tag)
+	}
+
+	// The call-site.
+	if d.opts.flag&(Lshortfile|Llongfile) != 0 {
```

Comment:
Oh, I wasn't even aware of this feature! So we could optionally turn this on in `lnd` with `WithCallerFlags`?

Reply:
yeah!